tests/nightly/model_backwards_compatibility_check/model_backwards_compat_train.py

```diff
@@ -0,0 +1,159 @@
+	upload_data_and_labels_to_s3(model_name)
+
+	inference_results = mod.predict(data_iter)
+	### Save inference_results
+	save_inference_results(inference_results, model_name)
+	### upload model and inference files to S3
+	files = list()
+	files.append(model_name + '-0001.params')
+	files.append(model_name + '-symbol.json')
+	files.append(model_name + '-inference')
+
+	mxnet_folder = str(mxnet_version) + backslash + model_name + backslash
+
+	upload_model_files_to_s3(files, mxnet_folder)
+
+	clean_model_files(files, model_name)
+
+def train_lenet_gluon_save_params_api():
+	model_name = 'lenet_gluon_save_params_api'
+	print ('Saving files for model %s' %model_name)
+	net = Net()
+	weights = mx.initializer.Xavier(magnitude = 2.57)
+	net.initialize(weights, ctx = [mx.cpu(0)])
+	### Prepare data
+
+	test_data = mx.nd.array(np.random.uniform(-1, 1, size=(20, 1, 30, 30)))
+	output = net(test_data)
+	# print (y)
+ #    ### Save the test data as well.
+ #    ### Save the inference output ys
+ #    ### Save the model params
+
+	mx.nd.save(model_name + '-data', {'data' : test_data})
+	save_inference_results(output, model_name)
+	net.save_params(model_name + '-params')
+
+	mxnet_folder = str(mxnet_version) + backslash + model_name + backslash
+
+	files = list()
+	files.append(model_name + '-data')
+	files.append(model_name + '-inference')
+	files.append(model_name + '-params')
+
+	upload_data_and_labels_to_s3(model_name)
+
+	upload_model_files_to_s3(files, mxnet_folder)
+
+	clean_model_files(files, model_name)
+
+def train_lenet_gluon_hybrid_export_api():
+	model_name = 'lenet_gluon_hybrid_export_api'
+	print ('Saving files for model %s' %model_name)
+	net = HybridNet()
+	weights = mx.initializer.Xavier(magnitude = 2.57)
+	net.initialize(weights, ctx = [mx.cpu(0)])
+	net.hybridize()
+	### Prepare data
+	test_data = mx.nd.array(np.random.uniform(-1, 1, size=(20, 1, 30, 30)))
+	output = net(test_data)
+	# print (y)
+    ### Save the test data as well.
+    ### Save the inference output ys
+    ### Save the model params
+
+	mx.nd.save(model_name + '-data', {'data' : test_data})
+	save_inference_results(output, model_name)
+	net.export(model_name, epoch=1)
+
+	mxnet_folder = str(mxnet_version) + backslash + model_name + backslash
+
+	files = list()
+	files.append(model_name + '-data')
+	files.append(model_name + '-inference')
+	files.append(model_name + '-0001.params')
+	files.append(model_name + '-symbol.json')
+
+
+	upload_data_and_labels_to_s3(model_name)
+
+	upload_model_files_to_s3(files, mxnet_folder)
+
+	clean_model_files(files, model_name)
+
+def train_lstm_gluon_save_parameters_api():
+	## If this code is being run on version >= 1.2.0 only then execute it, since it uses save_parameters and load_parameters API
+	if compare_versions(str(mxnet_version), '1.2.1')  < 0:
+		print ('Found MXNet version %s and exiting because this version does not contain save_parameters and load_parameters functions' %str(mxnet_version))
+		sys.exit(1)
```

Comment:
are we sure we want to exit instead of just returning?

Reply:
My bad. It should have been return. Will update it.

Reply:
Done.